src/main/scala/zio/prelude/AddMultiplyShape.scala

```diff
@@ -0,0 +1,100 @@
+package zio.prelude
+
+import zio.prelude.newtypes.{ Prod, Sum }
+
+trait AddMultiplyShape[A, +Addition[x] <: Associative[x], +Multiplication[x] <: Associative[x]] {
+
+  def add(l: => A, r: => A): A =
+    Sum.unwrap(Addition.combine(Sum(l), Sum(r)))
+
+  def multiply(l: => A, r: => A): A =
+    Prod.unwrap(Multiplication.combine(Prod(l), Prod(r)))
+
+  def Addition: Addition[Sum[A]]
```

Comment:
After looking at this, probably lower case `A` and `M` for `Addition` and `Multiplication` fields.

Reply:
That is definitely a reasonable proposition.
But then I don't know, how we will do it with lattices. Here, the operation ("addition") has a different name that the (imperative) method ("add"). But in a lattice, the operation and the method are both called "join".
So how do we resolve this?